query_execution/QueryExecutionTypedefs.hpp

```diff
@@ -65,7 +65,9 @@ enum QueryExecutionMessageType : message_type_id {
                               // their corresponding RelationalOperators.
   kRebuildWorkOrderMessage,  // From Foreman to Worker.
   kRebuildWorkOrderCompleteMessage,  // From Worker to Foreman.
-  kPoisonMessage,  // From the CLI shell to Foreman, then from Foreman to Workers.
+  kPoisonMessage,  // From the main thread to Foreman and workers.
+  kAdmitRequestMessage,  // Requesting a query (or queries) to be admitted.
+  kWorkloadCompletionMessage  // From Foreman to main thread.
```

Comment:
FYI, I sort these message types in the order of query life cycle. So, I think `kAdmitRequestMessage` could be in the first item, and `kWorkloadCompletionMessage` could put between `kRebuildWorkOrderCompleteMessage` and `kPoisonMessage`. Thoughts?